### PR TITLE
Feat: retrieval loop

### PR DESCRIPTION
**Pull request by fjebaker**

### CODE REVIEW

File: cart/rabbit.go [hunk 1]
```diff
@@ -90,11 +92,20 @@ func (rabbit *Rabbit) Update(t int32, player *Player, game *Game) {
...
 
 	if rabbit.PointInZone(player.X, player.Y) {
+		// does the player currently have the item
+		if player.HasItem {
+			rabbit.HappySfx.Play()
+			player.HasItem = false
+			game.NewDesiredItem()
+			rabbit.ItemSprite.Id = int32(game.DesiredItem)
+			rabbit.SetShowItem(t, false)
+		}
```
Comment (fjebaker) [inline cart/rabbit.go:102]: This is terrible praxis

File: cart/player.go [hunk 2]
```diff
@@ -144,6 +148,10 @@ func (player *Player) Update(t int32, world *World, game *Game) {
 		case world.IsItem(tileIndex):
 			world.CollectItem(x, y)
 		}
+		// check if the item was what we wanted
+		if tileIndex == int32(game.DesiredItem) {
```
Comment (JR-Mitchell) [inline cart/player.go:152]: Flagging that, once we add the possibility of desiring multi-block items (if we get to that point), this logic will need updating somehow

File: cart/rabbit.go (hunk 1 above)
Comment (JR-Mitchell) [reply cart/rabbit.go:102]: Yes

**Review by JR-Mitchell — APPROVED** [review]
GE4AGGJ

**Pull request merged**